Maths/ExponentialFunction.js

```diff
@@ -1,22 +1,23 @@
-let powerOfX = 1
-let factorial = 1
+import { calcFactorial } from './Factorial.js'
+
 /**
  * @function exponentialFunction
  * @description Calculates the n+1 th order Taylor series approximation of exponential function e^x given n
  * @param {Integer} power
  * @param {Integer} order - 1
- * @returns exponentialFunction(2,20) = 7.389056098930604
+ * @returns exponentialFunction(2,20) = 7.3890560989301735
  * @url https://en.wikipedia.org/wiki/Exponential_function
  */
 function exponentialFunction (power, n) {
+  let output = 0
   if (isNaN(power) || isNaN(n) || n < 0) {
     throw new TypeError('Invalid Input')
   }
   if (n === 0) { return 1 }
-  const recursion = exponentialFunction(power, n - 1)
-  powerOfX = powerOfX * power
-  factorial = factorial * n
-  return recursion + powerOfX / factorial
+  for(let i = 0; i < n; i++){
+    output += (power ** i) / calcFactorial(i)
```

Comment:
While it is a good idea to reuse algorithms in general, reusing the factorial implementation here is probably a bad idea because computing the factorial takes linear time, making the overall running time of your algorithm quadratic. I'd recommend simply keeping local variables `pow` and `fac` and updating them with each loop iteration to keep it linear.

Reply:
Right, I understand.